### DISCUSSION TITLE
Read redis and elasticsearch urls from environment variables

### DISCUSSION THREAD

**Discussion by henryclw** [ICIJ/datashare#903] [Ideas]
I hope we could read redis and elasticsearch urls from environment variables.

I want to run datashare in a custom docker compose file. But I've got other services using redis and elasitcsearch as well. So I don't want my redis container to have the name `redis`, I want to have the name `redis-datashare` or so. But now the redis url is fixed. (i.e. dee9e526ac94d4491114055a6b2beb806f1eb54c) Maybe it would be better if this is not hard coded? Say, reading it from environment variables, which could be passed by docker compose file.

**Comment by erikkri** [comment]
You can use the `--redisAddress` and `--elasticsearchAddress` options, which are documented on the server mode page: https://github.com/ICIJ/datashare/wiki/Datashare-Server-Mode#server-configuration

**Reply by henryclw** [reply]
Thank you for your kind reply, I'll try it
